src/Microsoft.Health.Fhir.SqlServer/Features/Storage/TvpRowGeneration/Merge/MergeSearchParameterRowGenerator.cs

```diff
@@ -0,0 +1,97 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using EnsureThat;
+using Microsoft.Health.Fhir.Core.Features.Search;
+using Microsoft.Health.SqlServer.Features.Schema.Model;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.Storage.TvpRowGeneration
+{
+    internal abstract class MergeSearchParameterRowGenerator<TSearchValue, TRow> : ITableValuedParameterRowGenerator<IReadOnlyList<MergeResourceWrapper>, TRow>
+        where TRow : struct
+    {
+        private readonly SearchParameterToSearchValueTypeMap _searchParameterTypeMap;
+        private readonly bool _isConvertSearchValueOverridden;
+        private bool _isInitialized;
+
+        protected MergeSearchParameterRowGenerator(SqlServerFhirModel model, SearchParameterToSearchValueTypeMap searchParameterTypeMap)
+        {
+            EnsureArg.IsNotNull(model, nameof(model));
+            EnsureArg.IsNotNull(searchParameterTypeMap, nameof(searchParameterTypeMap));
+
+            Model = model;
+            _searchParameterTypeMap = searchParameterTypeMap;
+            _isConvertSearchValueOverridden = GetType().GetMethod(nameof(ConvertSearchValue), BindingFlags.Instance | BindingFlags.NonPublic).DeclaringType != typeof(SearchParameterRowGenerator<TSearchValue, TRow>);
+        }
+
+        protected SqlServerFhirModel Model { get; }
+
+        public virtual IEnumerable<TRow> GenerateRows(IReadOnlyList<MergeResourceWrapper> resources)
```

Comment:
Add a check to ensure "resources" is not null.

Reply:
This is internal class.